far/macro.cpp

```diff
@@ -3912,7 +3912,6 @@ void FarMacroApi::editorsetFunc() const
 		ShowScrollBar           = 15,
 		EditOpenedForWrite      = 16,
 		SearchSelFound          = 17,
-		SearchRegexp            = 18,
```

Comment:
- Should I remove this option / index elsewhere, including documentation? I searched for `Regexp` in likely places, but I could miss some places altogether.
- What about backward compatibility? Will it break existing macros?

Reply:
I can do that. But why regexes only? How about case sensitive and fuzzy and the whole gamut of them?

Apparently, option `SearchPickUpWord` (19) was [removed from the code back in 2015](https://github.com/FarGroup/FarManager/commit/15d9f2fcade62d3c41b6cadb6a4032c5437dd6dc#diff-c125991267d8110ae22f0d586120f3c49298cb7bbe7a29fb4209b6f356a6c11c), but nobody updated the doc.

Maybe it's better to deprecate the option than maintaining some of parameters but not the others?

Up to you. Please let me know how to proceed.

Reply:
> But why regexes only? How about case sensitive and fuzzy and the whole gamut of them?

Unlike case etc. `[x] regex` usually changes the whole meaning of the search string, which sort of makes it more important.

However, I cannot immediately imagine a scenario when this might be useful. Let's remove and see how it goes, we can always put it back if something breaks. Could you remove 19 too please?

Reply:
Thanks. Done.